config-model/src/main/java/com/yahoo/schema/processing/AddDataTypeAndTransformToSummaryOfImportedFields.java

```diff
@@ -64,12 +64,7 @@ private static void setAttributeCombinerTransform(ImmutableSDField field, Summar
         if (summaryField.hasUnresolvedType()) {
             summaryField.setResolvedDataType(field.getDataType());
         }
-        if (summaryField.getTransform() == SummaryTransform.MATCHED_ATTRIBUTE_ELEMENTS_FILTER) {
-            // This field already has the correct transform.
-            return;
-        } else if (summaryField.getTransform() == SummaryTransform.MATCHED_ELEMENTS_FILTER) {
-            summaryField.setTransform(SummaryTransform.MATCHED_ATTRIBUTE_ELEMENTS_FILTER);
-        } else {
+        if (summaryField.getTransform() == SummaryTransform.NONE) {
```

Comment:
what if 'none of the above'?

Reply:
The code is structured to match the setAttributeTransform() member function: only set the transform if it has not yet been set (implied by it still being NONE).